pkg/volume/volume.go

```diff
@@ -132,6 +132,55 @@ type Unmounter interface {
 	TearDownAt(dir string) error
 }
 
+// BlockVolumeMapper interface provides methods to set up/map the volume.
+type BlockVolumeMapper interface {
+	Volume
+	// SetUpDevice prepares the volume to a self-determined directory path,
+	// which may or may not exist yet and returns combination of physical
+	// device path of a block volume and error.
+	// If the plugin is non-attachable, this method prepares the volume
+	// to be attached to a node. Also the non-attachable plugin needs to
+	// return proper physical device path.
+	// If the plugin is attachable, the volume has been attached already,
+	// just return empty string for device path.
+	// This may be called more than once, so implementations must be idempotent.
+	SetUpDevice() (string, error)
```

Comment:
From the description it's not clear to me what this function should do. If the plugin is attachable, it should not do anything, right? If the plugin is not attachable, it should prepare the device in /dev (or where appropriate) and return it's name.

Reply:
correct. I'll update the comment.